src/TemplateConverter.ump

```diff
@@ -0,0 +1,453 @@
+        }
+        
+        // If the final line of file contans only spaces, remove it completely
+            // And remove the newline before it (separately though)
+            // Because JET will not include it, but ours will
+            // Fixes constructor_StateMachineAssignDefault
+        if (line == null && containsOnlySpaces(previousLine)) {
+            // TODO: Make it more obvious what this depends on
+            int index = buffer.lastIndexOf(previousLine);
+            if (index >= 0 && !otherHadFunctionEnder && !(onlySpaces && previousOp == Operation.Code)) {
+                buffer.delete(index, buffer.length());
+                
+                // Then, need to delete the newline before it, if it exists
+                // But only if there wasn't a brace at the end
+                index = buffer.lastIndexOf(NewLine);
+                if (index != -1) {
+                    buffer.delete(index, buffer.length());
+                }
+            }
+            
+            // Should keep this in, since may indicate odd spacing
+            if (otherHadFunctionEnder && !previousLine.equals("")) {
+                // todo: isn't if it originally had a tab
+                printWarning("Has the odd end");
+            }
+        }
+        
+        if (line != null)
+        {
+            line = cutOffExtra(line);
+            
+            int indexOfNextPart = line.indexOf("<%");
+            String earlierPartOfLine = line.substring(0, indexOfNextPart);
+            line = line.substring(indexOfNextPart);
+            
+            buffer.append(earlierPartOfLine);
+            
+            // TODO: Explain the special case
+            // These should be based around ensuring that there are enough lines separating (increase to two)
+            // For when leading into a Code operation only
+            Operation nextOperation = getStartingOperation(line);
+            if (nextOperation == Operation.Code && previousLine != null) {
+                
+                // If leads into a <%, and there was a bracket on this line, with a blank line and a half, then add new line
+                if (otherHadFunctionEnder && containsOnlySpaces(previousLine) && containsOnlySpaces(earlierPartOfLine)) {
+                    buffer.append(NewLine);
+                }
+            }
+            
+            // If there are only two lines, one ending on the previous guy, and they are separated by spaces, then need to add extra.
+                // Occurs in JavaClassGenerator, line 183
+            if (nextOperation == Operation.Code && previousLine != null) {
+                String generated = buffer.toString();
+                int numNewlines = (generated.length() - generated.replace(NewLine, "").length()) / NewLine.length();
+                if (containsOnlySpaces(previousLine) && (previousOp == Operation.Code || previousOp == Operation.Directive || previousOp == Operation.Expression) && numNewlines == 2) {
```

Comment:
`previousOp == Operation.Code || previousOp == Operation.Directive || previousOp == Operation.Expression`
Could use an `EnumSet<Operation>` and just do `set.contains(previousOp)`. It's be nicer to read.